src/gpu/intel/ocl/tile_ops.h

```diff
@@ -614,11 +614,12 @@ __attribute__((overloadable)) void cooperative_prefetch_2d_internal(
         uint n_sg, uint sg_size, enum LSC_LDCC caching) {
     const uint cl_per_col = (rbytes + 63) >> 6;
     const uint cl = cl_per_col * c;
-    const uint cl_per_sg = (cl + n_sg - 1) / n_sg;
-    const uint cl_iters = (cl_per_sg + sg_size - 1) / sg_size;
+    const uint cl_per_sg = (cl + sg_size - 1) / sg_size;
+    const uint cl_iters = (cl_per_sg + n_sg - 1) / n_sg;
 #pragma unroll
     for (uint ii_cl = 0; ii_cl < cl_iters; ii_cl++) {
-        uint i_cl = ii_cl + (sg_id * cl_per_sg) + get_sub_group_local_id();
+        uint i_cl = ii_cl * cl_per_sg * sg_size + (sg_id * sg_size)
+                + get_sub_group_local_id();
```

Comment:
```suggestion
        uint i_cl = (ii_cl + (sg_id * cl_per_sg)) * sg_size + get_sub_group_local_id();
```

Reply:
Shouldn't it be:
```suggestion
        uint i_cl = (ii_cl * cl_per_sg + sg_id) * sg_size + get_sub_group_local_id();
```
Otherwise the second iteration will only offset by sg_size which will overlap with iteration zero of subgroup one.